builtin/credential/approle/path_login.go

```diff
@@ -54,6 +57,31 @@ func (b *backend) pathLoginUpdateAliasLookahead(ctx context.Context, req *logica
 	}, nil
 }
 
+func (b *backend) pathLoginResolveRole(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	// RoleID must be supplied during every login
+	roleID := strings.TrimSpace(data.Get("role_id").(string))
+	if roleID == "" {
+		return logical.ErrorResponse("missing role_id"), nil
+	}
+
+	// Look for the storage entry that maps the roleID to role
+	roleIDIndex, err := b.roleIDEntry(ctx, req.Storage, roleID)
+	if err != nil {
+		return nil, err
+	}
+	if roleIDIndex == nil {
+		return logical.ErrorResponse("invalid role ID"), nil
+	}
+
+	roleName := roleIDIndex.Name
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"role": roleName,
+		},
+	}, nil
+}
```

Comment:
It might make sense to create a small helper func like `logical.ResolveRoleResponse` that takes the role name and returns

```
&logical.Response{
    Data: map[string]interface{}{
        "role": roleName,
    },
}, nil
```

It's so basic but if we're introducing this to numerous auth methods it will provide consistency. What do you think?

Reply:
I like that idea a lot - especially as we'll be expecting all of the responses to be uniform. I'll make this change